addons/sale_timesheet/models/project.py

```diff
@@ -54,6 +54,36 @@ def _default_timesheet_product_id(self):
         default=_default_timesheet_product_id)
     warning_employee_rate = fields.Boolean(compute='_compute_warning_employee_rate')
     partner_id = fields.Many2one(compute='_compute_partner_id', store=True, readonly=False)
+    is_negative_margin = fields.Boolean('Negative Margin', compute="_compute_is_negative_margin", search="_search_is_negative_margin", groups="analytic.group_analytic_accounting")
+
+    @api.depends('analytic_account_id', 'task_ids', 'timesheet_ids')
+    def _compute_is_negative_margin(self):
+        profitability_data = self.env['project.profitability.report'].read_group([
+            ('project_id', 'in', self.ids)],
+            ['project_id', 'margin'], ['project_id'])
+        mapped_data = dict([(t['project_id'][0], t['margin']) for t in profitability_data])
+        for project in self:
+            project.is_negative_margin = bool(mapped_data.get(project.id, 0) < 0)
+
+    @api.model
+    def _search_is_negative_margin(self, operator, value):
+        if not isinstance(value, bool):
+            raise ValueError(_('Invalid value: %s') % value)
+        if operator not in ['=', '!=']:
+            raise ValueError(_('Invalid operator: %s') % operator)
+
+        query = """
+            SELECT P.id from project_project P
+                LEFT JOIN project_profitability_report R ON P.id = R.project_id
+                GROUP BY P.id
+                HAVING SUM(R.margin) < 0
+        """
+        if (operator == '=' and value is True) or (operator == '!=' and value is False):
+            operator_new = 'inselect'
+        else:
+            operator_new = 'not inselect'
+
+        return [('id', operator_new, (query, ()))]
```

Comment:
@xavierbol , Yes, we can calculate it over here,  but calculation requires the complex query from SOL, SO, product, project, and timesheet table. ([See here](https://github.com/odoo/odoo/blob/master/addons/sale_timesheet/report/project_profitability_report_analysis.py#L98)) and even for the project update view, we are using the same report for margin.  
so I think adding query will be a performance killer instead of getting the data from the report table. 

what do you think ?

Thanks

Reply:
This report is too slow when the db is big. So, I think this will be useless for that db.
Same for the query, I think we should do this when the report will be improved or changed. 🙂 
I discussed with our PO and we will do this point in another task.

Thanks for your understanding! 🙂

Reply:
Done!